influxdb/config.go

```diff
@@ -24,12 +24,15 @@ import (
 // MetricConfig defines the configuration file parameters for a sepcific metric
 // imported from InfluxDB.
 type MetricConfig struct {
-	Query          string
-	Database       string
-	Endpoint       string
-	Username       string
-	Password       string
-	TimeAggregated bool
+	Query    string `validate:"nonzero"`
+	Database string `validate:"nonzero"`
+	Endpoint string `validate:"nonzero"`
+	Auth     struct {
```

Comment:
Any reason for moving username & password into a separate substruct?

Reply:
Mainly to make it more obvious that they're a related concept -- if you don't want auth you can just not have the whole auth block when you're copying over the YAML example, instead of mentally having to ignore both the username and password. This might be more helpful when other forms are authentication (e.g. token) are introduced.

No strong opinions though, what do you think?

Reply:
I agree that it would have been useful if we had several authentication types, in which case we could have `BasicAuth` and `TokenAuth` substructs, with a check that only one of them is defined for a given metric. 

But since now we only have username & password, it feels like an unnecessary abstraction. I would probably suggest to keep everything on the same level for now.

Reply:
Makes sense, thanks!